cleanup_utils/mysql_cleanup.py

```diff
@@ -0,0 +1,136 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+MySQL Cleanup Script
+
+Cleans up all MySQL databases left behind from CI
+"""
+
+import logging
+import re
+from datetime import datetime
+from urllib.parse import urlparse
+
+import MySQLdb as mysql
+from MySQLdb import Error as MySQLError
+
+
+# Logging ####################################################################
+
+logger = logging.getLogger('integration_cleanup')
+
+
+# Functions ##################################################################
+
+def _get_cursor(url):
+    """
+    Returns a cursor on the database
+    """
+    database_url_obj = urlparse(url)
+    try:
+        connection = mysql.connect(
+            host=database_url_obj.hostname,
+            user=database_url_obj.username or '',
+            passwd=database_url_obj.password or '',
+            port=database_url_obj.port or 3306,
+        )
+    except MySQLError as exc:
+        logger.exception('Cannot get MySQL cursor: %s', exc)
+        raise
+    return connection.cursor()
+
+
+# Classes ####################################################################
+
+class MySqlCleanupInstance:
+    """
+    Handles the cleanup of old MySQL databases
+    """
+    def __init__(self, age_limit, url, domain, dry_run):
+        """
+        Set up variables needed for cleanup
+        """
+        self.age_limit = age_limit
+        self.cleaned_up_hashes = []
+        self.domain_suffix = domain.replace('.', '_')
+        self.dry_run = dry_run
+        try:
+            self.cursor = _get_cursor(url)
+        except Exception:  # pylint: disable=broad-except
+            # Exception has already been logged
+            self.cursor = None
+
+    def _get_old_databases(self):
+        """
+        Returns a list of databases older than the age limit
+        """
+        query = """SELECT table_schema, MAX(create_time) AS create_time
+            FROM information_schema.tables
+            WHERE create_time <= DATE_SUB(NOW(), INTERVAL %(age_limit)s DAY)
+            AND table_schema LIKE %(domain_filter)s
+            GROUP BY table_schema ORDER BY create_time DESC"""
+        params = {
+            'age_limit': self.age_limit,
+            'domain_filter': '%_{}_%'.format(self.domain_suffix)
+        }
+        self.cursor.execute(query, params)
```

Comment:
@jamestait, you have to wrap this in a `try-except` block and then log an error message if something goes wrong here.

Reply:
Done. In addition, I've returned an empty list in that instance, so the remainder of the cleanup will complete normally - we'll log the exception, log that no databases will be cleaned up, and the DNS cleanup will proceed as normal.